api/src/main/java/com/datatorrent/api/DAG.java

```diff
@@ -281,6 +306,20 @@
   public abstract OperatorMeta getMeta(Operator operator);
 
   /**
+   * Return all operators present in the DAG.
+   * @param <T>
+   * @return
+   */
+  public <T extends OperatorMeta> Collection<T> getAllOperators();
```

Comment:
Seems incomplete without having a way to get the rootOperators. If you want complete introspection you would need those to traverse the dag.